docs/i18n-docs.md

```diff
@@ -0,0 +1,187 @@
+<!--
+在开发Karmada dashboard的过程中，早期我们是通过人工的方式来维护locales文件，如果代码中新增了中文内容，则需要替换中文内容为对应的码点+i18n的调用函数，之后在中文的locales文件中插入码点和对应的中文内容的映射关系，再翻译中文内容，并在对应的语言的locales文件插入码点和对应翻译后内容的映射关系。在项目规模小，中文字符少的情况下手动维护的成本相对可以接受，但是一旦项目规模逐渐扩大、中文字符变多的情况下，手动维护locales文件的方式的成本会呈指数倍上升，因此需要一套i18n自动化工具来满足Karmada dashboard的i18n需求。在开源社区中我们发现很难找到一套开箱即用的i18n工具能够完美支持Karmada dashboard的i18n需求，因此我们希望从Karmada dashboard的i18n需求出发，实现一套满足Karmada dashboard的i18n需求的i18n自动化工具。
+-->
+
+### Goals
+
+1. Extract Chinese characters from the code and automatically replace them with calls to i18n functions
+2. Automatically maintain multilingual locales dictionary files.
+3. Support integration with third-party translation capabilities to automatically translate Chinese copy into the target language.
+
+<!--
+- 从代码中提取中文字符并自动将中文字符替换为i18n函数的调用
+- 自动地维护多语言的locales字典文件
+- 支持接入第三方翻译能力，自动的将中文文案翻译到目标语言
+-->
+
+### Non-Goals
+
+1. Developers need to manually add the styles of Chinese characters types that are not already supported by the tool when encountering them.
+2. Character length inconsistency encountered in multilingual display.
+
+<!--
+1. 开发人员需要在遇到工具不支持的汉字样式时手动添加样式。
+2. 多语言显示过程中遇到的字符长度不一致现象
+-->
+
+## Proposal
+
+<!--
+1. 自动节点识别:增强工具以自动识别和处理源代码中的各种节点类型，包括纯字符串节点、模板文字节点、JSX文本节点和JSX属性节点。此功能将确保对可能需要国际化的所有潜在文本元素进行全面扫描和修改。
+
+2. 汉字替换:实现一个健壮的机制，用相应的i18n密钥替换原始文件中所有检测到的汉字。这一步对于为多语言支持准备代码库和确保所有文本内容都为国际化做好准备至关重要。
+
+3. 与本地化文件的集成:自动将新的i18n键包含到locale /zh-CN中。json文件。此过程应有效地处理更新，确保所有密钥都准确地反映在本地化文件中，而无需人工干预。
+
+4. 翻译执行:开发和整合使用更新后的zh-CN的翻译程序。json文件生成对应的其他语言的翻译，如英语(en-US)和法语。该功能将允许快速和准确地制作多语言内容。
+-->
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a developer of the Karmada dashboard, when introducing new features with a significant amount of new Chinese content, there is an automated tool that can extract, translate, and generate corresponding locale files for this new content with a single click, thereby reducing the maintenance cost of i18n. The community administrator provides the corresponding i18n translation rules for specific descriptions. In automatic translation, you can preferentially use the i18n translation rules provided by the community administrator and combine them with automatic translation to improve the overall effect of i18n translation.
```

Comment:
It seems that, it's two user-stories?

Reply:
I think it's same situation to community members.